src/GafferImage/DisplayTransform.cpp

```diff
@@ -132,27 +116,22 @@ void DisplayTransform::hashTransform( const Gaffer::Context *context, IECore::Mu
 	h.append( colorSpace );
 	h.append( display );
 	h.append( view );
-	h.append( contextName );
-	h.append( contextValue );
-}
 
-OpenColorIO::ConstContextRcPtr DisplayTransform::getLocalContext(OpenColorIO::ConstConfigRcPtr config) const
-{
-
-	OpenColorIO::ConstContextRcPtr context = config->getCurrentContext();
-	OpenColorIO::ContextRcPtr mutableContext;
-	std::string contextName = contextNamePlug()->getValue();
-	std::string contextValue = contextValuePlug()->getValue();
+	std::string name;
+	std::string value;
 
-	if(!contextName.empty() && !contextValue.empty())
+	for( CompoundDataPlug::MemberPlugIterator it( p ); !it.done(); ++it )
 	{
-		if(!mutableContext) mutableContext = context->createEditableCopy();
-		mutableContext->setStringVar(contextName.c_str(), contextValue.c_str());
+		IECore::DataPtr d = p->memberDataAndName( it->get(), name );
+		if( d )
+		{	
+			value = runTimeCast<StringData>(d)->readable();
```

Comment:
this is my personal Everest! ( it's not forbidden to smile from that comment )
More seriously, it takes a bit to figure out how to play with DataPtr when you don't know and you struggle to find an example.